modules/Login/src/server/routes/loginRoutes.coffee

```diff
@@ -47,11 +34,28 @@ exports.setupRoutes = (app, passport) ->
 	app.get '/login/ssoFailure', exports.ssoFailure
 
 
+exports.getRedirectUrl = (req) ->
+	redirectUrl = null
+	
+	# Check to see if the RelayState value is set in the request
+	if req.body?.RelayState? && req.body.RelayState != ""
+		redirectUrl = req.body.RelayState
+		console.log "redirecting to #{redirectUrl}"
+	else
+		parsedUrl = url.parse(req.originalUrl || req.url)
```

Comment:
I'm having trouble understanding what this does, even after reading the docs https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost

Would help to provide some examples here of what the overall URL vs the pathname vs the path will produce

Reply:
It's a legacy API but it supports relative paths as opposed to the new WHATWG URL standard they added to node which forces URLs into a format which requires a host. I used it because its a standard nodejs library and gives me a quick way to grab the host, path and query parameters of a string. 

There is this nice diagram explaining what the pathname, host and path keys are on the object:

<img width="824" alt="Screen Shot 2022-07-19 at 4 51 22 PM" src="https://user-images.githubusercontent.com/868119/179867035-be3e1a26-99b2-4474-afd5-c37ef3bdb925.png">